### ISSUE TITLE
`Oversample` and `AntiAlias`

### ISSUE THREAD

**Issue by Jordan-Dennis** [LouisDesdoigts/dLux#78]
Hi all,
I have been giving quite a lot of thought to anti-aliasing and oversampling. A good starting place for the anti-aliasing is the "classic el-cheapo" averaging over an over-sampled pixel grid (hence the over-sampling), to quote @benjaminpope. I was thinking that we could add `oversample` as a method/attribute to the `fft` based propagators, but I understand that @LouisDesdoigts already has a system (using `interpolate_paraxial` in `Wavefront`). My motivation is that we have broken the `get_pixel_positions` of the `Wavefront` and into `coordiantes.py` and are looking to do something similar. If the `Propagators` automatically perform the oversampling it could simplify the work later on as well as being a little more explicit.

**Comment by benjaminpope** [comment]
Ok @Jordan-Dennis  I think we can do a pretty easy differentiable oversample of edges using `sigmoid` from `jax.nn`: I have added an example [here](https://github.com/LouisDesdoigts/dLux/blob/soft/notebooks/soft_edged_aperture.ipynb) on branch `soft` doing this for a circular aperture. This should generalize straightforwardly to any binary aperture defined by an inequality (which is most of the ones we think about). 

We can probably use this for differentiably modelling the HST NICMOS aperture.

**Comment by LouisDesdoigts** [comment]
This was resolved by the Apertures merge.
